packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -216,6 +233,9 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
 
         this.initializeLinkHover();
 
+        this.showCommandSeparator = this.preferences['terminal.integrated.enableCommandSeparator'] ?? false;
+        this.initializeOSC133Support();
```

Comment:
Its fine to do this check this when the Terminal is created, but imho we also need a listener on the preference and update this value then according to the preference. Otherwise the setting only has effect when the user creates a new terminal. This makes it hard for the user to decide which option they like more.